src/Grassmann.jl

```diff
@@ -249,8 +253,33 @@ function __init__()
             generate_derivation(:(Reduce.Algebra),T,:df,:RExpr)
         end
     end
-    @require SymPy="24249f21-da20-56a4-8eb1-6a02cf4ae2e6" generate_algebra(:SymPy,:Sym,:diff,:symbols)
-    @require SymEngine="123dc426-2d89-5057-bbad-38513e3affd8" generate_algebra(:SymEngine,:Basic,:diff,:symbols)
+    @require SymPy="24249f21-da20-56a4-8eb1-6a02cf4ae2e6" begin
+        generate_algebra(:SymPy,:Sym,:diff,:symbols)
+        T = Union{TensorGraded, TensorMixed}
+        symbolic = v -> typeof(v) == SymPy.Sym
+        SymPy.expand(x::T) = map(v -> symbolic(v) ? SymPy.expand(v) : v, x)
+        SymPy.factor(x::T) = map(v -> symbolic(v) ? SymPy.factor(v) : v, x)
+        SymPy.together(x::T) = map(v -> symbolic(v) ? SymPy.together(v) : v, x)
+        SymPy.apart(x::T) = map(v -> symbolic(v) ? SymPy.apart(v) : v, x)
+        SymPy.cancel(x::T) = map(v -> symbolic(v) ? SymPy.cancel(v) : v, x)
+        # SymPy.expand_trig(x::T) = map(v -> symbolic(v) ? SymPy.expand_trig(v) : v, x)
+        SymPy.N(x::T, args...) = map(v -> symbolic(v) ? SymPy.N(v, args...) : v, x)
+        # SymPy.evalf(X::T, args...) = map(v -> symbolic(v) ? SymPy.evalf(v, args...) : v, x)
+        SymPy.convert(X::T, args...) = map(v -> symbolic(v) ? SymPy.convert(v, args...) : v, x)
+        SymPy.subs(x::T, args...) = map(v -> symbolic(v) ? SymPy.subs(v, args...) : v, x)
+        SymPy.collect(x::T, args...) = map(v -> symbolic(v) ? SymPy.collect(v, args...) : v, x)
+
+    end
+    @require SymEngine="123dc426-2d89-5057-bbad-38513e3affd8" begin
+        generate_algebra(:SymEngine,:Basic,:diff,:symbols)
+        T = Union{TensorGraded, TensorMixed}
+        symbolic = v -> typeof(v) == SymEngine.Basic
+        SymEngine.expand(x::T) = map(v -> symbolic(v) ? SymEngine.expand(v) : v, x)
+        SymEngine.subs(x::T, args...) = map(v -> symbolic(v) ? SymEngine.subs(v, args...) : v, x)
+        SymEngine.N(x::T) = map(v -> symbolic(v) ? SymEngine.N(v) : v, x)
+        SymEngine.evalf(X::T, args...) = map(v -> symbolic(v) ? SymEngine.evalf(v, args...) : v, x)
+        SymEngine.convert(X::T, args...) = map(v -> symbolic(v) ? SymEngine.convert(v, args...) : v, x)
```

Comment:
This is type piracy, you are hijacking the global `convert` method for `TensorGraded` and `TensorMixed`.

Type piracy like this is not allowed, could you explain what you're trying to accomplish?

Reply:
I just wanted to provide an interface to `SymEngine.jl`'s `convert` method. Maybe this is type piracy by extension?

Reply:
Instead of `TensorGraded` or `TensorMixed`, try `Chain{V,G,SymEngine.Basic} where {V,G}` or `MultiVector{V,SymEngine.Basic} where V`. This method should only apply when you have `Sym` as the type, but not in general.

You are over-riding the global `convert` methods. You need to restrict yourself to a subset of elements that is actually relevant to what you are trying to accomplish.